libyul/backends/evm/StackLayoutGenerator.cpp

```diff
@@ -540,6 +572,41 @@ void StackLayoutGenerator::stitchConditionalJumps(CFG::BasicBlock const& _block)
 			},
 			[&](CFG::BasicBlock::FunctionReturn const&)	{},
 			[&](CFG::BasicBlock::Terminated const&) { },
+			[&](CFG::BasicBlock::Switch const& _switch)
+			{
+				Stack exitLayout = info.exitLayout;
+				// The last block must have produced the switch expression at the stack top.
+				yulAssert(!exitLayout.empty(), "");
+				yulAssert(exitLayout.back() == _switch.switchExpr, "");
+				// The switch expression is consumed by the switch.
+				//exitLayout.pop_back();
+
+				auto fixJumpTargetEntry = [&](Stack const& _originalEntryLayout) -> Stack {
+					Stack newEntryLayout = exitLayout;
+					// Whatever the block being jumped to does not actually require, can be marked as junk.
+					for (auto& slot: newEntryLayout)
+						if (!util::contains(_originalEntryLayout, slot))
+							slot = JunkSlot{};
+					// Make sure everything the block being jumped to requires is actually present or can be generated.
+					for (auto const& slot: _originalEntryLayout)
+						yulAssert(canBeFreelyGenerated(slot) || util::contains(newEntryLayout, slot), "");
+					return newEntryLayout;
+				};
+
+				auto& defaultCaseInfo = m_layout.blockInfos.at(_switch.defaultCase);
+				defaultCaseInfo.entryLayout = fixJumpTargetEntry(defaultCaseInfo.entryLayout);
+				_addChild(_switch.defaultCase);
+				for (auto const& [caseValue, caseBlock]: _switch.cases)
+				{
+					auto& caseInfo = m_layout.blockInfos.at(caseBlock);
+					caseInfo.entryLayout = fixJumpTargetEntry(caseInfo.entryLayout);
+					_addChild(caseBlock);
+				}
+
+				_addChild(_switch.defaultCase);
+				for (auto const& [caseValue, caseBlock]: _switch.cases)
+					_addChild(caseBlock);
```

Comment:
This is redundant

Reply:
Fixed in [f4c11dd](https://github.com/ethereum/solidity/pull/12978/commits/f4c11dd29942f1deaa45a32fb04be2a92caa5599)